cpg-language-java/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/java/JavaLanguage.kt

```diff
@@ -57,18 +57,34 @@ open class JavaLanguage :
     override val conjunctiveOperators = listOf("&&")
     override val disjunctiveOperators = listOf("||")
 
+    /**
+     * See
+     * [Java Language Specification](https://docs.oracle.com/javase/specs/jls/se19/html/jls-4.html).
+     */
     @Transient
     @JsonIgnore
     override val simpleTypes =
         mapOf(
-            "boolean" to IntegerType("boolean", 1, this, NumericType.Modifier.SIGNED),
+            // Boolean Types:
+            // https://docs.oracle.com/javase/specs/jls/se19/html/jls-4.html#jls-4.2.5
+            "boolean" to BooleanType("boolean", 1, this),
+            "Boolean" to BooleanType("java.lang.Boolean", 1, this),
+            "java.lang.Boolean" to BooleanType("java.lang.Boolean", 1, this),
+
+            // Integral Types:
```

Comment:
Should we also add `java.lang.Integer` and such here? They also "behave" like their respective primitive types.

Reply:
Same as above

Reply:
Yes